lib/ansible/modules/remote_management/cpm/cpm_portconfig.py

```diff
@@ -0,0 +1,304 @@
+    serial_logoff: "^H"
+"""
+
+RETURN = """
+data:
+    description: The output JSON returned from the commands sent
+    returned: always
+    type: str
+"""
+
+import base64
+import json
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils._text import to_text, to_bytes, to_native
+from ansible.module_utils.six.moves.urllib.error import HTTPError, URLError
+from ansible.module_utils.urls import open_url, ConnectionError, SSLValidationError
+
+
+def assemble_json(cpmmodule):
+    json_load = '{"serialports":{"port": "%s"' % to_native(cpmmodule.params["serial_port"])
+
+    if cpmmodule.params["serial_portname"] is not None:
+        json_load = '%s,"portname": "%s"' % (json_load, to_native(cpmmodule.params["serial_portname"]))
+    if cpmmodule.params["serial_baud"] is not None:
+        json_load = '%s,"baud": %s' % (json_load, to_native(cpmmodule.params["serial_baud"]))
+    if cpmmodule.params["serial_handshake"] is not None:
+        json_load = '%s,"handshake": %s' % (json_load, to_native(cpmmodule.params["serial_handshake"]))
+    if cpmmodule.params["serial_stopbits"] is not None:
+        json_load = '%s,"stopbits": %s' % (json_load, to_native(cpmmodule.params["serial_stopbits"]))
+    if cpmmodule.params["serial_parity"] is not None:
+        json_load = '%s,"parity": %s' % (json_load, to_native(cpmmodule.params["serial_parity"]))
+    if cpmmodule.params["serial_mode"] is not None:
+        json_load = '%s,"mode": %s' % (json_load, to_native(cpmmodule.params["serial_mode"]))
+    if cpmmodule.params["serial_cmd"] is not None:
+        json_load = '%s,"cmd": %s' % (json_load, to_native(cpmmodule.params["serial_cmd"]))
+    if cpmmodule.params["serial_seq"] is not None:
+        json_load = '%s,"seq": %s' % (json_load, to_native(cpmmodule.params["serial_seq"]))
+    if cpmmodule.params["serial_tout"] is not None:
+        json_load = '%s,"tout": %s' % (json_load, to_native(cpmmodule.params["serial_tout"]))
+    if cpmmodule.params["serial_echo"] is not None:
+        json_load = '%s,"echo": %s' % (json_load, to_native(cpmmodule.params["serial_echo"]))
+    if cpmmodule.params["serial_break"] is not None:
+        json_load = '%s,"break": %s' % (json_load, to_native(cpmmodule.params["serial_break"]))
+    if cpmmodule.params["serial_logoff"] is not None and (len(cpmmodule.params["serial_logoff"]) > 0):
+        json_load = '%s,"logoff": "%s"' % (json_load, to_native(cpmmodule.params["serial_logoff"]))
+
+    json_load = '%s}}' % (json_load)
+    return json_load
+
+
+def run_module():
+    # define the available arguments/parameters that a user can pass to
+    # the module
+    module_args = dict(
+        cpm_action=dict(choices=['getport', 'setport'], required=True),
+        cpm_url=dict(type='str', required=True),
+        cpm_username=dict(type='str', required=True),
+        cpm_password=dict(type='str', required=True, no_log=True),
+        serial_port=dict(type='int', required=True),
```

Comment:
I think all the `serial_` prefixes are repetitive. I think this module should be renamed `cpm_serial_port` then you can just remove all the `serial_` prefixes. This would also remove confusion on whether this module is configuring serial or network ports.